### PR TITLE
ENH: allow SlidingEstimator and gat to get **fit_params at fit

### PR DESCRIPTION
**Pull request by kingjr**
such that one can do e.g.

```python
clf = SlidingEstimator(LinearModel())
clf.fit(X, y, sample_weights=np.random.rand(len(X))
```

### CODE REVIEW

**Comment by agramfort** [comment]
Use directly sample_weights and avoid kwargs

**Comment by kingjr** [comment]
> Use directly sample_weights and avoid kwargs

will break if estimator is a pipeline. 

This is sklearn API

**Comment by kingjr** [comment]
I *think* Travis' errors are unrelated (but perhaps should be fixed? I don't understand them)

**Comment by larsoner** [comment]
They are `sklearn` errors that should be fixed. The 3.6 runs use sklearn 0.19 whereas the 2.7 runs use 0.18.1 so it's probably a scikit-learn API problem. I can try to look this week unless @agramfort beats me to it

File: doc/whats_new.rst [hunk 1]
```diff
@@ -85,6 +85,8 @@ Changelog
 
     - Add function :func:`mne.channels.get_builtin_montages` to list all built-in montages by `Clemens Brunner`_
 
+    - :class:`mne.decoding.SlidingEstimator` and :class:`mne.decoding.GeneralizingEstimator` now accept **fit_params at fitting by `Jean-Remi King`_
```
Comment (larsoner) [inline doc/whats_new.rst]: better to put double-backticks on `**fit_params` so it's not errantly interpreted as emphasis

File: mne/decoding/search_light.py [hunk 2]
```diff
@@ -65,6 +65,8 @@ def fit(self, X, y):
             X.shape = (n_samples, n_features_1, n_features_2, n_tasks)
         y : array, shape (n_samples,) | (n_samples, n_targets)
             The target values.
+        **fit_params : dict of string -> object
```
Comment (larsoner) [inline mne/decoding/search_light.py]: just `dict` is fine here, the fact that they get passed is clear in the description

**Comment by agramfort** [comment]
indeed  LGTM then.

+1 for MRG if travis failure is unrelated

**Comment by larsoner** [comment]
Thanks @kingjr

**Pull request merged**